qemu/tests/qemu_guest_agent.py

```diff
@@ -1450,6 +1450,70 @@ def gagent_check_get_cpustats(self, test, params, env):
             test.fail("Number of cpus is not correct, cpu_num_qga is: %d"
                       "cpu_num_guest is %d" % (cpu_num_qga, int(cpu_num_guest-1)))
 
+    def gagent_check_get_diskstats(self, test, params, env):
+        """
+        Execute "guest-get-diskstats" command to guest agent.
+
+        Steps:
+        1) Check numbers of disk argumentss info
+        2) Check disk stats info.
+        3) Check disk numbers.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        session = self._get_session(params, self.vm)
+        self._open_session_list.append(session)
+
+        ds_info_qga = self.gagent.get_diskstats()
+
+        error_context.context("Check diskstats argument numbers.", LOG_JOB.info)
+        # check the number of arguments whether is correct
+        # with official document
+        num_arg_guest = int(session.cmd_output(params['count_num_arg']))
+        key_list = params["diskstats_info_list"].split(",")
+        num_arg_def = len(list(key_list))
+        if num_arg_guest != num_arg_def:
+            test.error("Diskstats argument numbers may change, "
+                       "please take a look.")
+
+        disk_num_qga = 0
+        improper_list = []
+        for ds in ds_info_qga:
+            ds_name = ds['name']
+            error_context.context("Check %s 'major' and 'minor' value." % ds_name,
+                                  LOG_JOB.info)
+            ds_major_qga = int(ds['major'])
+            ds_minor_qga = int(ds['minor'])
+            ds_major_guest = int(session.cmd_output(params["cmd_ds_major"] % ds_name))
+            ds_minor_guest = int(session.cmd_output(params["cmd_ds_minor"] % ds_name))
+
+            if ds_major_qga != ds_major_guest or ds_minor_qga != ds_minor_guest:
+                test.error("Major's or minor's value is not correct, "
+                           "please take a look.")
```

Comment:
Again, please output the values when it is not expected, also here should be `fail` as it is your key check point and it does not meet test expectation.

Reply:
> Again, please output the values when it is not expected, also here should be `fail` as it is your key check point and it does not meet test expectation.

this part has been updated.

Reply:
> I would recommend to output all the info together here when this checkpoint failed, as the case won't continue and other values won't be checked then, we at least could check them together in the log.

actually, the whole output info is the output from the command 'guest-get-diskstats', and the contrasting part had 
 been written to the cfg file. So, do we still need to print cfg file content in log?

Reply:
> > I would recommend to output all the info together here when this checkpoint failed, as the case won't continue and other values won't be checked then, we at least could check them together in the log.
> 
> actually, the whole output info is the output from the command 'guest-get-diskstats', and the contrasting part had been written to the cfg file. So, do we still need to print cfg file content in log?

Then you will need to scroll up and down when you check the log, and the information is distributed everywhere, I am afraid that way you will not check them one by one after you got failure, why not make it more convenient?

Reply:
okay, updated this part and tested it with triggering error, info looks like:

[stdlog] 2023-03-22 04:50:08,493 avocado.test ERROR| avocado.core.exceptions.TestError: Major's or minor's value is not correct, major&minor in guest are: 8, 0by qga are: 8, 0. Since the following checkpoints are not detected, print all cfg file contents here: name,read-sectors,read-ios,read-merges,write-sectors,write-ios,write-merges,discard-sectors,discard-ios,discard-merges,flush-ios,read-ticks,write-ticks,discard-ticks,flush-ticks,ios-pgr,total-ticks,weight-ticks,major,minor